v0/destinations/klaviyo/transform.js

```diff
@@ -13,84 +11,33 @@ const {
   CONFIG_CATEGORIES,
   BASE_ENDPOINT,
   MAPPING_CONFIG,
-  LIST_CONF,
   ecomExclusionKeys,
   ecomEvents,
   eventNameMapping,
   jsonNameMapping
 } = require("./config");
+const {
+  isProfileExist,
+  checkForMembersAndSubscribe,
+  createCustomerProperties
+} = require("./util");
 const {
   defaultRequestConfig,
   constructPayload,
   getFieldValueFromMessage,
-  defaultGetRequestConfig,
   defaultPostRequestConfig,
   extractCustomFields,
-  removeUndefinedValues,
   toUnixTimestamp,
   removeUndefinedAndNullValues,
   getSuccessRespEvents,
   getErrorRespEvents,
   CustomError,
   isEmptyObject,
   addExternalIdToTraits,
-  adduserIdFromExternalId
+  adduserIdFromExternalId,
+  defaultPutRequestConfig
 } = require("../../util");
 
-// A sigle func to handle the addition of user to a list
-// from an identify call.
-// DOCS: https://www.klaviyo.com/docs/api/v2/lists
-const addUserToList = async (message, traitsInfo, conf, destination) => {
-  // Check if list Id is present in message properties, if yes override
-  let targetUrl = `${BASE_ENDPOINT}/api/v2/list/${destination.Config.listId}`;
-  if (get(traitsInfo.properties, "listId")) {
-    targetUrl = `${BASE_ENDPOINT}/api/v2/list/${get(
-      traitsInfo.properties,
-      "listId"
-    )}`;
-  }
-  let profile = {
-    id: getFieldValueFromMessage(message, "userId"),
-    email: getFieldValueFromMessage(message, "email"),
-    phone_number: getFieldValueFromMessage(message, "phone")
-  };
-  if (destination.Config.enforceEmailAsPrimary) {
-    delete profile.id;
-    profile._id = getFieldValueFromMessage(message, "userId");
-  }
-  // If func is called as membership func else subscribe func
-  if (conf === LIST_CONF.MEMBERSHIP) {
-    targetUrl = `${targetUrl}/members`;
-  } else {
-    // get consent statuses from message if availabe else from dest config
-    targetUrl = `${targetUrl}/subscribe`;
-    profile.sms_consent = get(traitsInfo.properties, "smsConsent")
-      ? get(traitsInfo.properties, "smsConsent")
-      : destination.Config.smsConsent;
-    profile.$consent = get(traitsInfo.properties, "consent")
-      ? get(traitsInfo.properties, "consent")
-      : destination.Config.consent;
-  }
-  profile = removeUndefinedValues(profile);
-  // send network request
-  try {
-    await axios.post(
-      targetUrl,
-      {
-        api_key: destination.Config.privateApiKey,
-        profiles: [profile]
-      },
-      {
-        headers: {
-          "Content-Type": "application/json"
-        }
-      }
-    );
-  } catch (err) {
-    logger.debug("[Klaviyo :: addToList]", err);
-  }
-};
-
 // ---------------------
 // Main Identify request handler func
```

Comment:
CAn we make this inline docs more aligned to the changes we are making

Reply:
done